### DISCUSSION TITLE
Users Leaving Servers open, blocking others on Genre Classical Jamulus Server Space

### DISCUSSION THREAD

**Discussion by jm98029** [jamulussoftware/jamulus#1202] [General]
People using the Classical genre on jamulus are leaving their servers registered and on.  This keeps others from registering a server and using jamulus. 

I have tried numerous times to register a server on that genre, only to find it full, with no one logged into the servers (0/10).

Can this be made to stop, so that other can use the space?

Thank you for looking into this.

**Comment by dcorson-ticino-com** [comment]
Why don't you just use one of those empty servers?

**Comment by jm98029** [comment]
I could, but I am not happy when others interlope on my server. 
Isn't this an abuse of the system?


    On Sunday, March 7, 2021, 02:14:51 PM PST, DonC <notifications@github.com> wrote:  
 
 


Why don't you just use one of those empty servers?

**Reply by gene96817** [reply]
I am seeing several scenarios with contradictions.

1- Here is a public server. Anyone can use it. I only use the server a few hours a week. The server is a community asset. (I thought this part of the community ethic.)

2- I have a server that is only for my use. I don't want anybody else using my server. (In which case why put it on a public listing.)

3- I see a publicly listed server. Am I trespassing if I use the server? (No, that is why it is publicly listed.)

4- I have a public server. My rehearsals are being intruded by trolls. Maybe I need to make my server private and invisible.  (This is sad. Maybe we need to solve this problem.)

5- My server is publicly listed but I really intended it to be used by my small circle of friends. This is why my server name has "private" appended to the name. The directory makes it easy for me and my friends to use our server.  Everybody, please stay out. (I don't expect the public directory to be a convenience for operating a private server.)

**Reply by jujudusud** [reply]
https://jamulus.io/wiki/Running-a-Server
In this section of the manual, you can read how to use a server with the members of your band only. It is the way it has to be used.

**Comment by gene96817** [comment]
I can't tell you how often my server is used. Officially less than 12 scheduled hours per week of use. Taking a peek frequently, I am delighted to see others using my server.

**Comment by dcorson-ticino-com** [comment]
>Isn't this an abuse of the system?

Not at all.  Everyone who has a server listed knows that it is a public resource.  That is why it is called a public server.
I too have a public server that is used 3 evenings a week by me and I am very happy to see it being used by others more and more.

Note that if you are worried that others could bother you use the solo button.  You only hear the users for whom solo is set.  So set solo for all the people you are playing with and then you will not hear anyone else who may stop in.
It is even possible for 2 different groups to play on the same server at the same time without bothering each other with this method when everyone solos the people they are playing with.

**Reply by Hk1020** [reply]
> > Isn't this an abuse of the system?
> 
> Not at all. Everyone who has a server listed knows that it is a public resource. 

I am not sure about that. I think many people believe they need a server because of this:

> 5- My server is publicly listed but I really intended it to be used by my small circle of friends. This is why my server name has "private" appended to the name. The directory makes it easy for me and my friends to use our server. Everybody, please stay out. (I don't expect the public directory to be a convenience for operating a private server.)

And if it isn’t listed people can’t join.

Not true we know but it is rather hard to communicate the private address otherwise. You‘d want your own domain or have to rely on potentially unstable IP addresses. 

What people really want is a space to do their regular meetings without anyone else coming in. Just as in real life. They don’t want a server.

**Reply by jujudusud** [reply]
https://jamulus.io/wiki/Running-a-Server
In this section of the manual, you can read how to use a server with the members if your band only. It is the way it has to be used.

**Comment by dakhubgit** [comment]
There is absolutely no need to have your server listed in order to have a closed group of people use it.  All you have to do is make sure that your server is able to be reached on its Jamulus port, usually 22124 UDP.  Your server can have just an IP address, but you can use a DynDNS service to give it a name, too.  Server providers usually also offer reserving domains but that tends to pin you to a particular provider in some regard.  I actually set up and dismantle a server on a weekly base.  It gets a different IP address every time but using a DynDNS service, everybody reaches it under the same name.  I have even swapped out servers in mid-session because one was not up to the task (for the first few trials I tended to have backup servers up in parallel with "safe" defaults).

My server is not registered publicly because, well, this is a closed ensemble and the servers are a bit expensive to have them running for anything but the rehearsals themselves.  I do have one on permanent rent as emergency backup and for experiments.  I may let it slide eventually.

At any rate, the public registries are for servers that are intended to be shared.  If that's not your intent, I'd argue _that_ is an abuse of this resource because you are blocking a listed server slot without actually offering a server to the public.

**Comment by jimallard** [comment]
I just recently started a server in the classical space for my choir. I registered it as a public server to make it easier to find for some of my elderly, technology challenged choirmates. I have a hard time getting them to type anything correctly, so this approach seemed natural.

Is there an actual limit on the number of public servers listed, or is the issue that the list gets crowded and your server is to find?

I've set up dynamic DNS with noip, and have a host name, so I could go the private server route. Since I'm just starting, what's the typical practice for large groups of 40+ elderly participants?

**Reply by ghost** [reply]
Maybe there could be a new central server reserved for listing temporary servers (maybe time limited to 1 day).
Especially as the central server space gets used up.

**Reply by Hk1020** [reply]
> 
> 
> Maybe there could be a new central server reserved for listing temporary servers (maybe time limited to 1 day).
> Especially as the central server space gets used up.

This sounds like a very good idea. But we'd have to think about how to reregister once disconnected.

**Reply by ann0see** [reply]
> Maybe there could be a new central server reserved for listing temporary servers (maybe time limited to 1 day).

As far as I know, something like that should already exist. Corrados created a custom directory server for this purpose.

You could also set up a central server/directory server and let them type the domain into the custom central server box. Afterwards let them select the custom central server box in the genre list of Jamulus. Also, since Jamulus saves the last servers you connected to, it should be a one time setup? Would this work for you?

**Comment by dakhubgit** [comment]
ann0see <notifications@github.com> writes:

>> Maybe there could be a new central server reserved for listing
> temporary servers (maybe time limited to 1 day).
>
> As far as I know, something like that should already exist. Corrados
> created a custom directory server for this purpose.
>
> You could also set up a central server/directory server and let them
> type the domain into the custom central server box. Afterwards let
> them select the custom central server box in the genre list of
> Jamulus. Also, since Jamulus saves the last servers you connected to,
> it should be a one time setup? Would this work for you?

I can affirm that all our ensemble members ever do is click "connect"
and "disconnect", with the server name being kept once you got it right.

-- 
David Kastrup
